app/vmctl/remoteread/remoteread.go

```diff
@@ -173,8 +178,15 @@ func (c *Client) Ping() error {
 
 func (c *Client) fetch(ctx context.Context, data []byte, streamCb StreamCallback) error {
 	r := bytes.NewReader(data)
-	url := c.addr + remoteReadPath
-	req, err := http.NewRequest(http.MethodPost, url, r)
+	u, err := url.JoinPath(c.addr, c.path)
+	if err != nil {
+		return fmt.Errorf("error create url from addr %s and path %s", c.addr, c.path)
+	}
+	// use default path to remote read protocol if path is empty
+	if u == c.addr {
+		u = c.addr + remoteReadPath
+	}
```

Comment:
Should we remove this logic in favour of setting default value of `remoteReadSrcPath ` flag to `api/v1/read`?

Reply:
logic updated